test/cppsuite/tests/reverse_split.cpp

```diff
@@ -0,0 +1,105 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "src/common/constants.h"
+#include "src/common/logger.h"
+#include "src/main/test.h"
+
+namespace test_harness {
+/*
+ * This test inserts data at the end of the collection and truncates off from the start of the
+ * collection. In doing so pages at the start of the tree are gradually emptied while pages are
+ * added at the end of the tree. This means the test frequently executes the reverse split path.
+ */
+class reverse_split : public test {
+    public:
+    reverse_split(test_args &args) : test(args)
+    {
+        /* Add split timing stresses to the conn_open config. */
+        if (args.wt_open_config == "") {
+            std::string stress;
+            if (random_generator::instance().generate_bool())
+                stress = "timing_stress_for_test=[split_3]";
+            else
+                stress = "timing_stress_for_test=[split_4]";
+            logger::log_msg(LOG_WARN, "Adding config to WiredTiger open: " + stress);
+            args.wt_open_config = stress;
+        }
+        init_operation_tracker();
+    }
+
+    /* Remove operation simulates burst truncates. */
+    void
+    remove_operation(thread_worker *tc) override
+    {
+        logger::log_msg(
+          LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+        /* Must have unique collection for each thread. */
+        testutil_assert(tc->db.get_collection_count() == tc->thread_count);
+        collection &coll = tc->db.get_collection(tc->id);
+        scoped_cursor write_cursor = std::move(tc->session.open_scoped_cursor(coll.name));
+
+        while (tc->running()) {
+            /* Synchronise across our truncate threads so they don't diverge over time. */
+            tc->sync();
+            testutil_check(write_cursor->reset(write_cursor.get()));
+            tc->txn.begin();
+            int ret = write_cursor->next(write_cursor.get());
+            if (ret != 0) {
+                tc->txn.rollback();
+                continue;
+            }
+            const char *key;
+            testutil_check(write_cursor->get_key(write_cursor.get(), &key));
+            const std::string key_str(key);
+            uint64_t min_key_id = std::stoi(key_str);
+            uint64_t key_count = coll.get_key_count();
+            /* Truncate up to 80% of the range. */
+            uint64_t end_key_id = random_generator::instance().generate_integer<uint64_t>(
+              min_key_id, min_key_id + ((key_count - min_key_id) / 1.2));
```

Comment:
Dividing by 1.2 will give slightly more that 80% (about 83%).

Reply:
87c1d8abf